File: platform/polyglot/confluence_api/py/src/confluence_api/models/color_scheme.py
"""Color scheme models for Confluence look-and-feel customisation."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, ConfigDict, Field

from confluence_api.logger import create_logger

log = create_logger('confluence_api', __file__)


class ColorSchemeModel(BaseModel):
    """Individual color scheme defining UI element colors."""

    model_config = ConfigDict(populate_by_name=True)

    top_bar: str = Field("", alias="topBar")
    top_bar_menu_selected_background: str = Field(
        "", alias="topBarMenuSelectedBackground"
    )
    top_bar_menu_selected_text: str = Field("", alias="topBarMenuSelectedText")
    top_bar_menu_item_text: str = Field("", alias="topBarMenuItemText")
    header_button_base_bg: str = Field("", alias="headerButtonBaseBg")
    header_button_base_text: str = Field("", alias="headerButtonBaseText")
    heading_text: str = Field("", alias="headingText")
    links: str = ""
    border_color: str = Field("", alias="borderColor")
    nav_background: str = Field("", alias="navBackground")


class ColorSchemeThemeBasedModel(BaseModel):
    """Color scheme tied to a specific theme."""

    model_config = ConfigDict(populate_by_name=True)

    theme_key: str = Field("", alias="themeKey")
    color_scheme: ColorSchemeModel | None = Field(None, alias="colorScheme")


class SpaceColorSchemeTypeModel(BaseModel):
    """Indicates whether a space uses a custom or global color scheme."""

    model_config = ConfigDict(populate_by_name=True)

    type: str = ""

File: platform/polyglot/confluence_api/mjs/src/models/color-scheme.mjs
/**
 * @module models/color-scheme
 * @description Color scheme and theme schemas for the Confluence API.
 */
import { z } from 'zod';

/**
 * Color scheme model defining the primary UI colors for a Confluence
 * space or global theme.
 */
export const ColorSchemeModelSchema = z.object({
  /** Background color of the top navigation bar. */
  topBar: z.string().default(''),
  /** Background color of the selected item in the top bar menu. */
  topBarMenuSelectedBackground: z.string().default(''),
  /** Text color of the selected item in the top bar menu. */
  topBarMenuSelectedText: z.string().default(''),
  /** Text color of items in the top bar menu. */
  topBarMenuItemText: z.string().default(''),
  /** Base background color for header buttons. */
  headerButtonBaseBg: z.string().default(''),
  /** Base text color for header buttons. */
  headerButtonBaseText: z.string().default(''),
  /** Color used for heading text. */
  headingText: z.string().default(''),
  /** Color used for hyperlinks. */
  links: z.string().default(''),
  /** Color used for borders and dividers. */
  borderColor: z.string().default(''),
  /** Background color of the left-side navigation panel. */
  navBackground: z.string().default(''),
});

/**
 * Color scheme paired with its parent theme key.
 */
export const ColorSchemeThemeBasedModelSchema = z.object({
  /** Key identifying the parent theme (e.g. "default", "documentation"). */
  themeKey: z.string().default(''),
  /** Color scheme overrides for the specified theme. */
  colorScheme: ColorSchemeModelSchema.optional(),
});

/**
 * Type descriptor indicating whether a space uses a custom or inherited
 * color scheme.
 */
export const SpaceColorSchemeTypeModelSchema = z.object({
  /** Type of color scheme assignment (e.g. "custom", "theme"). */
  type: z.string().default(''),
});
